### PR TITLE
CRS-193: add types to DateSeparator

### PR DESCRIPTION
**Pull request by jaapbakker88**
# Submit a pull request

## CLA

- [ ] I have signed the [Stream CLA](https://docs.google.com/forms/d/e/1FAIpQLScFKsKkAJI7mhCr7K9rEIOpqIDThrWxuvxnwUq2XkHyG154vQ/viewform) (required).
- [ ] The code changes follow best practices
- [ ] Code changes are tested (add some information if not applicable)

### CODE REVIEW

File: src/components/DateSeparator/DateSeparator.js [hunk 1]
```diff
@@ -7,9 +8,15 @@ import { withTranslationContext } from '../../context';
  * DateSeparator - A simple date separator
  *
  * @example ../../docs/DateSeparator.md
+ * @type {React.FC<import('types').DateSeparatorProps>}
  */
-const DateSeparator = ({ position, tDateTimeParser, formatDate, date }) => {
-  if (!Date.parse(date)) return null;
+const DateSeparator = ({
+  position = 'right',
```
Comment (vishalnarkhede) [inline src/components/DateSeparator/DateSeparator.js:14]: does the styleguidist pick these default values?

Comment (jaapbakker88) [reply src/components/DateSeparator/DateSeparator.js:14]: Good call, not sure.

Comment (vishalnarkhede) [reply src/components/DateSeparator/DateSeparator.js:14]: If not, then lets not remove `DateSeparator.defaultProps` part

Comment (jaapbakker88) [reply src/components/DateSeparator/DateSeparator.js:14]: image

picks them up

Comment (vishalnarkhede) [reply src/components/DateSeparator/DateSeparator.js:14]: Noice!!

**Pull request merged**